### ISSUE TITLE
Proposal: Official GopherJS Docker images

### ISSUE THREAD

**Issue by flimzy** [gopherjs/gopherjs#820]
I wonder if others would find GopherJS docker images to be useful, similar to the official [Golang docker images](https://hub.docker.com/_/golang/). (Apparently [at least one other person](https://github.com/xxuejie/docker-gopherjs) found the idea useful).

This would help me, for one, with the CI/CD stages for my various GopherJS projects.  I would see such an image would have Go, node.js, and GopherJS (with syscall and source map support) installed.

If we decide this is a worthy goal, I would be happy to take on the project.There would be a few logistical concerns to straighten out (management of a dockerhub (or other registry) account, where the official Dockerfile(s) should live, etc).

**Comment by flimzy** [comment]
I have an image I'm using: https://hub.docker.com/r/flimzy/gopherjs/

It serves my purposes, but if it would be useful to others, it would probably be useful to extend it further, by using proper version tags, and possibly develop an 'alpine' version, etc.

I'm just using the Docker Hub automated build capabilities, which are pretty limited with their tagging support, so doing something in a Circle/Travis-CI script to push images on update would likely be in order.

I'll wait to see if others have an interest in this before I invest that effort, though.

**Comment by roscopecoltran** [comment]
Hi guys,

Hope you are all well !

I think that a slim docker image would be awesome, as it could be really useful any developers giving a try to gopherjs and share it.

**Quick feedback from my last week-end on gopherjs:**
I spent my week-end to create a slim docker image, based on alpine 3.8.1, trying to create a stable docker image, that I could share with some dev contacts, more specifically to demo [jsgo](https://github.com/dave/jsgo)...

And, I thought that it would be a quick personal task, for making a turnkey setup for gopherjs, and sharing a docker compose, with an extra-contrib allowing to mount a [docker-volume-git](https://github.com/kassisol/docker-volume-git), all of these good intentions, happened to be quiet messy because I had to debug many things due to alpine apk packages specificity, [musl](https://alpinelinux.org/posts/Alpine-Linux-has-switched-to-musl-libc.html) vs [glibc](https://github.com/sgerrand/alpine-pkg-glibc), syscall without much up to date guidance...

Being late to deliver something shareable and slim, one of my contact started to be sarcastic as he could not find a repo named "awesome-gopherjs", nor any showcases or stacks with gopherjs+vue, gopherjs+react and gopherjs+angularjs for real-world use case but only small proof of concepts. He summarized the discussion like it sounds like a technical artistry project...

So, please find below a list of suggestions/projects, CMS related, that could fit well with gopherjs with the most popular js frameworks:

**Showcase suggestions:**
- https://github.com/akveo/ngx-admin for angular web-admin interface
- https://github.com/marmelab/react-admin for react web-admin interface
- https://github.com/PanJiaChen/vue-element-admin for vuejs web-admin interface
- https://github.com/PanJiaChen/vueAdmin-template
   - https://github.com/PanJiaChen/electron-vue-admin
- https://github.com/YuraDev/vue-chrome-extension-template for web/chrome/electron frontends
   - https://github.com/ALiangLiang/vue-webpack-chrome-extension-template
- https://github.com/cucygh/vue-chrome-extension-example
- https://github.com/hybridgroup/gocv for exciting stuff around Augmented/Mixed Reality using a webcam and a web frontend

**Gopherjs packaging**
- **Please, at least,** Create an homebrew formula :-), for mac users, with a valid go version and make [syscalls](https://github.com/gopherjs/gopherjs/blob/master/doc/syscalls.md) working on osx (I still have a warning in my chrome browser despite following the doc)
- Create more real world showcases/boilerplates, like mentioned without forecasting advanced used of graphql and grpc for a clone like [graphcms](https://graphcms.com/), even a wordpress like  or any admin web-interface (a simple cms with a front and backend) would allow to forsee the benefits of gopherjs, and trigger a community traction.

So, unfortunately, I agree with the post above some how, more real-world efforts are required... As, it is quiet hard to identify gopherjs as a "nice to have project" or a high community traction project ? Would it end up like Google plus ? How safe is the bet ?

In a nutshell, I will still to continue to play with jsgo, as the work if for a smart personal dev playground, but I think that people need to figure out the SWOT for gopherjs.

Cheers,
Rosco
